diff --git a/configure.ac b/configure.ac
index 9505f56..d671f5a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,5 +1,5 @@
 AC_PREREQ([2.60])
-AC_INIT([DRI2Proto], [2.4], [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg])
+AC_INIT([DRI2Proto], [2.6], [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg])
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
 AM_MAINTAINER_MODE
 
diff --git a/dri2proto.h b/dri2proto.h
index ff76355..cd82afb 100644
--- a/dri2proto.h
+++ b/dri2proto.h
@@ -296,7 +296,8 @@ typedef struct {
     CARD32 ust_lo B32;
     CARD32 msc_hi B32;
     CARD32 msc_lo B32;
-    CARD32 sbc B32;
+    CARD32 sbc_hi B32;
+    CARD32 sbc_lo B32;
 } xDRI2BufferSwapComplete;
 #define sz_xDRI2BufferSwapComplete 32
 
@@ -304,6 +305,21 @@ typedef struct {
     CARD8 type;
     CARD8 pad;
     CARD16 sequenceNumber B16;
+    CARD16 event_type B16;
+    CARD16 pad2;
+    CARD32 drawable B32;
+    CARD32 ust_hi B32;
+    CARD32 ust_lo B32;
+    CARD32 msc_hi B32;
+    CARD32 msc_lo B32;
+    CARD32 sbc B32;
+} xDRI2BufferSwapComplete2;
+#define sz_xDRI2BufferSwapComplete2 32
+
+typedef struct {
+    CARD8 type;
+    CARD8 pad;
+    CARD16 sequenceNumber B16;
     CARD32 drawable B32;
     CARD32 pad1 B32;
     CARD32 pad2 B32;
diff --git a/dri2proto.txt b/dri2proto.txt
index dc46e58..df763c7 100644
--- a/dri2proto.txt
+++ b/dri2proto.txt
@@ -178,7 +178,8 @@ DRI2ATTACHMENT { DRI2BufferFrontLeft
 	       	 DRI2BufferAccum
 		 DRI2BufferFakeFrontLeft
 		 DRI2BufferFakeFrontRight
-		 DRI2BufferDepthStencil }
+		 DRI2BufferDepthStencil
+		 DRI2BufferHiz }
 
 	These values describe various attachment points for DRI2
 	buffers.
@@ -509,6 +510,8 @@ The DRI2 extension has undergone a number of revisions before
 
 	2.3: Added the DRI2InvalidateBuffers event.
 
+	2.6: Enlightenment attained.  Added the DRI2BufferHiz attachment.
+
 Compatibility up to 2.0 is not preserved, but was also never released.
 
 
@@ -569,6 +572,7 @@ A.1 Common Types
 	0x7	DRI2BufferFakeFrontLeft
 	0x8	DRI2BufferFakeFrontRight
 	0x9	DRI2BufferDepthStencil
+	0xa	DRI2BufferHiz
 └───
 	Used to encode the possible attachment points.  The attachment
 	DRI2BufferDepthStencil is only available with protocol version 1.1 or
diff --git a/dri2tokens.h b/dri2tokens.h
index 7804e4d..16c9008 100644
--- a/dri2tokens.h
+++ b/dri2tokens.h
@@ -43,6 +43,7 @@
 #define DRI2BufferFakeFrontLeft		7
 #define DRI2BufferFakeFrontRight	8
 #define DRI2BufferDepthStencil		9
+#define DRI2BufferHiz			10
 
 #define DRI2DriverDRI			0
 #define DRI2DriverVDPAU			1
